travis/setup_dependencies_common.sh

```diff
@@ -135,20 +135,22 @@ if [[ ! -z $CONDA_DEPENDENCIES ]]; then
 fi
 
 # NUMPY
+# We use --no-pin to avoid installing other dependencies just yet.
+
 if [[ $NUMPY_VERSION == dev* ]]; then
     # We install nomkl here to make sure that Numpy and Scipy versions
     # installed subsequently don't depend on the MKL. If we don't do this, then
     # we run into issues when we install the developer version of Numpy
     # because it is then not compiled against the MKL, and one runs into issues
     # if Scipy *is* still compiled against the MKL.
-    conda install $QUIET nomkl
+    conda install $QUIET --no-pin nomkl
     # We then install Numpy itself at the bottom of this script
     export CONDA_INSTALL="conda install $QUIET python=$PYTHON_VERSION"
 elif [[ $NUMPY_VERSION == stable ]]; then
-    conda install $QUIET numpy=$LATEST_NUMPY_STABLE
+    conda install $QUIET --no-pin numpy=$LATEST_NUMPY_STABLE
```

Comment:
Will it not be the case that when installing the main dependencies later, there will still be the issue that the pin file can contain ``numpy stable*``?